pkg/pipeline/decode/decode_protobuf.go

```diff
@@ -37,36 +39,76 @@ func PBFlowToMap(flow *pbflow.Record) config.GenericMap {
 		return config.GenericMap{}
 	}
 	out := config.GenericMap{
-		"FlowDirection":          int(flow.Direction.Number()),
-		"Bytes":                  flow.Bytes,
-		"SrcAddr":                ipToStr(flow.Network.GetSrcAddr()),
-		"DstAddr":                ipToStr(flow.Network.GetDstAddr()),
-		"SrcMac":                 macToStr(flow.DataLink.GetSrcMac()),
-		"DstMac":                 macToStr(flow.DataLink.GetDstMac()),
-		"SrcPort":                flow.Transport.GetSrcPort(),
-		"DstPort":                flow.Transport.GetDstPort(),
-		"Etype":                  flow.EthProtocol,
-		"Packets":                flow.Packets,
-		"Duplicate":              flow.Duplicate,
-		"Proto":                  flow.Transport.GetProtocol(),
-		"TimeFlowStartMs":        flow.TimeFlowStart.AsTime().UnixMilli(),
-		"TimeFlowEndMs":          flow.TimeFlowEnd.AsTime().UnixMilli(),
-		"TimeReceived":           time.Now().Unix(),
-		"Interface":              flow.Interface,
-		"AgentIP":                ipToStr(flow.AgentIp),
-		"Flags":                  flow.Flags,
-		"IcmpType":               flow.GetIcmpType(),
-		"IcmpCode":               flow.GetIcmpCode(),
-		"TcpDropBytes":           flow.TcpDropBytes,
-		"TcpDropPackets":         flow.TcpDropPackets,
-		"TcpDropLatestFlags":     flow.GetTcpDropLatestFlags(),
-		"TcpDropLatestState":     tcpStateToStr(flow.GetTcpDropLatestState()),
-		"TcpDropLatestDropCause": tcpDropCauseToStr(flow.GetTcpDropLatestDropCause()),
-		"DnsRequestTimeMs":       flow.TimeDnsReq.AsTime().UnixMilli(),
-		"DnsResponseTimeMs":      flow.TimeDnsRsp.AsTime().UnixMilli(),
-		"DnsId":                  flow.GetDnsId(),
-		"DnsFlags":               flow.GetDnsFlags(),
-		"TimeFlowRttMs":          flow.TimeFlowRtt.AsDuration().Milliseconds(),
+		"FlowDirection":   int(flow.Direction.Number()),
+		"Bytes":           flow.Bytes,
+		"SrcAddr":         ipToStr(flow.Network.GetSrcAddr()),
+		"DstAddr":         ipToStr(flow.Network.GetDstAddr()),
+		"SrcMac":          macToStr(flow.DataLink.GetSrcMac()),
+		"DstMac":          macToStr(flow.DataLink.GetDstMac()),
+		"Etype":           flow.EthProtocol,
+		"Packets":         flow.Packets,
+		"Duplicate":       flow.Duplicate,
+		"Proto":           flow.Transport.GetProtocol(),
+		"TimeFlowStartMs": flow.TimeFlowStart.AsTime().UnixMilli(),
+		"TimeFlowEndMs":   flow.TimeFlowEnd.AsTime().UnixMilli(),
+		"TimeReceived":    time.Now().Unix(),
+		"Interface":       flow.Interface,
+		"AgentIP":         ipToStr(flow.AgentIp),
+	}
+
+	proto := flow.Transport.GetProtocol()
+	if proto == syscall.IPPROTO_ICMP || proto == syscall.IPPROTO_ICMPV6 {
+		icmpOut := config.GenericMap{
+			"IcmpType": flow.GetIcmpType(),
+			"IcmpCode": flow.GetIcmpCode(),
+		}
+		out = out.Append(icmpOut)
```

Comment:
out of curiosity: why here and everywhere, you don't add the fields to `out` directly?

Reply:
no real reason I switched to direct assign to save the memory used with map allocation and removed Append method